components/email.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * Manage BuddyPress Email Post Types.
+ *
+ * @since 1.6.0
+ */
+class BPCLI_Email extends BPCLI_Component {
+	/**
+	 * Create a new email post connected to an email type.
+	 *
+	 * ## OPTIONS
+	 *
+	 * --type=<type>
+	 * : Email type for the email (should be unique identifier, sanitized like a post slug).
+	 *
+	 * [--type-description=<type-description>]
+	 * : Email type description.
+	 *
+	 * --subject=<subject>
+	 * : Email subject line. Email tokens allowed. View https://codex.buddypress.org/emails/email-tokens/ for more info.
+	 *
+	 * --content=<content>
+	 * : Email content. Email tokens allowed. View https://codex.buddypress.org/emails/email-tokens/ for more info.
+	 *
+	 * [--plain-text-content=<plain-text-content>]
+	 * : Plain-text email content. Email tokens allowed. View https://codex.buddypress.org/emails/email-tokens/ for more info.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp email create --type=new-event --type-description="Send an email when a new event is created" --subject="[{{{site.name}}}] A new event was created" --content="<a href='{{{some.custom-token-url}}}'></a>A new event</a> was created" --plain-text-content="A new event was created"
+	 *     Success: Email post created for type "new-event".
+	 *
+	 * @alias add
+	 */
+	public function create( $args, $assoc_args ) {
+		$switched = false;
+
+		if ( false === bp_is_root_blog() ) {
+			$switched = true;
+			switch_to_blog( bp_get_root_blog_id() );
+		}
+
+		$term = term_exists( $assoc_args['type'], bp_get_email_tax_type() );
+
+		// Term already exists so don't do anything.
+		if ( $term !== 0 && $term !== null ) {
+			if ( true === $switched ) {
+				restore_current_blog();
+			}
+
+			WP_CLI::error( "Email type '{$assoc_args['type']}' already exists." );
+		}
+
+		if ( ! empty( $args[0] ) ) {
```

Comment:
There is no field in the doc that will use `$args[0]`. All the fields have a `--command=<content>` schema.

Reply:
Yeah, this needs documentation, but it looks like a good feature - I was thinking when reading the examples how tedious it'd be to type the entire `content` at the command line :) 

If `content` comes from a file, then the `--content` flag should no longer be required, right? This means that the documentation should be changed, and the requirement of either `--content` or a content file should be enforced within the method.

Reply:
I've fixed this in commit 382cd27.

See related discussion here - https://github.com/buddypress/wp-cli-buddypress/pull/57#discussion_r168215442

Reply:
Got it - this seems better.